[TensorExpr] LoopNest: add a constructor that takes Stmt instead of list of Tensors.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45949 [TensorExpr] LoopNest: add a constructor that takes Stmt instead of list of Tensors.**
* #45948 [TensorExpr] LoopNest: replace output_tensors_ with output_bufs_.
* #45947 [TensorExpr] Loopnest: unify intermediate_tensors_ and temp_bufs_.
* #45946 [TensorExpr] Add shorthand versions for `splitWith{Mask,Tail}` functions.
* #45936 [TensorExpr] Nuke `Function` class and directly use `Tensor` instead.

Differential Revision: [D24156001](https://our.internmc.facebook.com/intern/diff/D24156001)

Commit:
[TensorExpr] LoopNest: add a constructor that takes Stmt instead of list of Tensors.

[ghstack-poisoned]

Commit:
rebase on "[TensorExpr] LoopNest: add a constructor that takes Stmt instead of list of Tensors."


Differential Revision: [D24156001](https://our.internmc.facebook.com/intern/diff/D24156001)

[ghstack-poisoned]

nickgg commented:
buf_initializers should not be necessary after lowering to stmt, feels like you could safely remove that.

Commit:
Update on "[TensorExpr] LoopNest: add a constructor that takes Stmt instead of list of Tensors."


Differential Revision: [D24156001](https://our.internmc.facebook.com/intern/diff/D24156001)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] LoopNest: add a constructor that takes Stmt instead of list of Tensors."


Differential Revision: [D24156001](https://our.internmc.facebook.com/intern/diff/D24156001)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] LoopNest: add a constructor that takes Stmt instead of list of Tensors."


Differential Revision: [D24156001](https://our.internmc.facebook.com/intern/diff/D24156001)

[ghstack-poisoned]
